gammapy/spectrum/tests/test_spectralgrouping.py

```diff
@@ -0,0 +1,87 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals)
+from astropy.coordinates import SkyCoord, Angle
+from numpy.testing import assert_allclose
+from ...utils.energy import EnergyBounds
+from ...image import ExclusionMask
+from ...data import DataStore
+from ...region import SkyCircleRegion
+from ...datasets import gammapy_extra
+from ...spectrum import SpectrumExtraction
+from ...spectrum.spectral_grouping import SpectralGrouping
+from ...spectrum.spectrum_fit import SpectrumFit
+from astropy.tests.helper import assert_quantity_allclose
+
+def make_spectrum_extraction():
+    # Construct w/o config file
+    center = SkyCoord(83.63, 22.01, unit='deg', frame='icrs')
+    radius = Angle('0.3 deg')
+    on_region = SkyCircleRegion(pos=center, radius=radius)
+
+    bkg_method = dict(type='reflected', n_min=2)
+
+    exclusion_file = gammapy_extra.filename(
+        "datasets/exclusion_masks/tevcat_exclusion.fits")
+    excl = ExclusionMask.from_fits(exclusion_file)
+
+    bounds = EnergyBounds.equal_log_spacing(1, 10, 40, unit='TeV')
+
+    obs = [23523, 23559, 23526, 23592]
+    store = gammapy_extra.filename("datasets/hess-crab4-hd-hap-prod2")
+    ds = DataStore.from_dir(store)
+
+    ana = SpectrumExtraction(datastore=ds, obs_ids=obs, on_region=on_region,
+                             bkg_method=bkg_method, exclusion=excl,
+                             ebounds=bounds)
+    return ana
+
+def test_define_spectral_groups():
```

Comment:
When running this locally without gammapy-extra I get this error

```
GammapyExtraNotFoundError: The gammapy-extra repo is not available. You have to set the GAMMAPY_EXTRA environment variable to point to the location for it to be found.
```

You have to mark all test that require gammapy-extra with
``
@requires_data('gammapy-extra')
``
see e.g.
https://github.com/gammapy/gammapy/blob/master/gammapy/spectrum/tests/test_spectrum_extraction.py#L17

The same holds for optional dependencies (such as sherpa, and YAML)